### PR TITLE
Refs [TASK][36452] Category data source

### PR DESCRIPTION
**Pull request by VietKFC**
## Related Tickets
- [#Task 36452: Get categories data source](https://edu-redmine.sun-asterisk.vn/issues/36452)
## WHAT
- Categories data source
## Review Checklist

Category | View Point | Description | Self review | Reviewer2 (name)
--- | --- | --- | --- | ---
Conventions | Sun* coding conventions | https://github.com/framgia/coding-standards/tree/master/eng/android |<li>- [ ] yes</li>|<li>- [ ] yes</li>
Conventions | Kotlin coding conventions | https://kotlinlang.org/docs/coding-conventions.html |<li>- [ ] yes</li>|<li>- [ ] yes</li>
Redmine | Sun* Redmine working process  | https://github.com/framgia/Training-Guideline/blob/master/WorkingProcess/redmine/redmine.md |<li>- [ ] yes</li>|<li>- [ ] yes</li>

## Notes (Optional)
*(Impacted Areas in Application(List features, api, models or services that this PR will affect))*

*(List gem, library third party add new)*

*(Other notes)*

### CODE REVIEW

File: app/src/main/java/com/example/cocktailmaster/ui/category/CategoryPresenter.kt [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+import android.content.Context
+import com.example.cocktailmaster.data.model.Category
+import com.example.cocktailmaster.data.repository.CategoryRemoteRepository
+import com.example.cocktailmaster.data.source.remote.utils.RequestAPICallback
+import com.example.cocktailmaster.ui.CategoryUtils
+
+class CategoryPresenter(
+    private val categoryView: CategoryContract.View,
+    private val categoryRepo: CategoryRemoteRepository,
+    private val context: Context
```
Comment (thonc-2135) [inline app/src/main/java/com/example/cocktailmaster/ui/category/CategoryPresenter.kt]: trong presenter k nên chưa context

File: app/src/main/java/com/example/cocktailmaster/ui/category/CategoryPresenter.kt [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+                    it.categoryThumb = CategoryUtils.getCategoryMap(it.name , context)
+                }
+                categoryView.showCategories(data)
+                categoryView.hideLoading()
+            }
+
+            override fun onFailed() {
+                categoryView.showError()
+            }
+
```
Comment (thonc-2135) [inline app/src/main/java/com/example/cocktailmaster/ui/category/CategoryPresenter.kt]: remove

File: app/src/main/java/com/example/cocktailmaster/ui/category/CategoryPresenter.kt [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+            override fun onSuccess(data: List<Category>) {
+                data.forEach {
+                    it.categoryThumb = CategoryUtils.getCategoryMap(it.name , context)
+                }
+                categoryView.showCategories(data)
+                categoryView.hideLoading()
+            }
+
+            override fun onFailed() {
+                categoryView.showError()
```
Comment (thonc-2135) [inline app/src/main/java/com/example/cocktailmaster/ui/category/CategoryPresenter.kt:21]: error là tự hide loading à

File: app/src/main/java/com/example/cocktailmaster/ui/category/CategoryPresenter.kt (hunk 1 above)
Comment (VietKFC) [reply app/src/main/java/com/example/cocktailmaster/ui/category/CategoryPresenter.kt]: em hỏi chút ,nếu trong presenter , datasource , repository mà mình muốn dùng context thì truyền vào thế nào ạ

File: app/src/main/java/com/example/cocktailmaster/ui/category/CategoryPresenter.kt (hunk 3 above)
Comment (VietKFC) [reply app/src/main/java/com/example/cocktailmaster/ui/category/CategoryPresenter.kt:21]: em da them roi a

**Pull request merged**